src/snowflake/snowpark/column.py

```diff
@@ -303,6 +303,16 @@ def __bool__(self) -> bool:
             "'~' for 'not' if you're building DataFrame filter expressions. For example, use df.filter((col1 > 1) & (col2 > 2)) instead of df.filter(col1 > 1 and col2 > 2)."
         )
 
+    def __iter__(self) -> None:
+        raise TypeError(
+            "Column is not iterable. This error can occur when you use the Python built-ins for sum, min and max. Please make sure you use the corresponding function from snowflake.snowpark.functions."
+        )
+
+    def __round__(self, n=None):
```

Comment:
We override `__pow__` on `Column`. Maybe we should support `__round__` too. It's not an aggregate function.
Let's discuss tmr.

Reply:
Discussed that we'll only support `__pow__`.